src/components/Shape.tsx

```diff
@@ -52,23 +73,48 @@ class Shape extends React.Component<FigureProps, any> {
             const x3 = this.props.x + size / 2;
             const y3 = this.props.y;
             const points = `${x1},${y1} ${x2},${y2} ${x3},${y3}`;
-            return <polygon onMouseDown={onMouseDown}
+            return <polygon onMouseDown={onMouseDown} onClick={onClick}
                             ref="element" points={points} fill="#fff" stroke="#000"/>;
         }
-        else if (this.props.type == 'circle')
-            return <ellipse onMouseDown={onMouseDown}
-                            ref="element" cx={this.props.x} cy={this.props.y} rx={size/2} ry={size/2} fill="#fff"
+        else if (this.props.type == 'circle') {
+            const r = size / 2;
+            return <ellipse onMouseDown={onMouseDown} onClick={onClick}
+                            ref="element" cx={this.props.x+r} cy={this.props.y+r} rx={r} ry={r} fill="#fff"
                             stroke="#000"/>;
-        else
+        } else
             return (
-                <rect onMouseDown={onMouseDown}
+                <rect onMouseDown={onMouseDown} onClick={onClick}
                       ref="element"
                       width={size}
                       height={size}
                       x={this.props.x}
                       y={this.props.y} fill="#fff" stroke="#000"/>
             );
     }
+
+    renderSelectionPoint(ref: string, x: number, y: number) {
+        return <rect ref={ref} x={x - 2} y={y - 2} width={4} height={4} fill="#000" stroke="#000" />
+    }
+
+    // svg does not support style.transform
+    render() {
+        this.dragElements = ['element'];
+
+        if (this.props.selected) {
+            this.dragElements = this.dragElements.concat('select_rect', 'p1', 'p2', 'p3', 'p4');
+
+            return <g>
+                <rect ref='select_rect' x={this.props.x} y={this.props.y} width={size} height={size} fill="#fff" stroke="#000" />
+                {this.renderSelectionPoint('p1', this.props.x, this.props.y)}
+                {this.renderSelectionPoint('p2', this.props.x+size, this.props.y)}
+                {this.renderSelectionPoint('p3', this.props.x, this.props.y+size)}
+                {this.renderSelectionPoint('p4', this.props.x+size, this.props.y+size)}
```

Comment:
i would recommend creating a separate component for the selection box, and putting it in it's own layer that's above all the other components. this way the user can see all the resize handles and has access to them even if there's a shape above the selected item.

![screen shot 2016-07-14 at 10 50 34 am](https://cloud.githubusercontent.com/assets/37429/16849752/cf36668e-49b0-11e6-8329-e3bbf108f0cd.png)